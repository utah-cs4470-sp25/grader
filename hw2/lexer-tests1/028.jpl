// OK

struct rect {
  W : int
  H : int
}

let r = rect{ 800, 600 }

show r.W
show r.H
