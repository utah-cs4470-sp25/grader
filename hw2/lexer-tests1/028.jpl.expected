NEWLINE
STRUCT 'struct'
VARIABLE 'rect'
LCURLY '{'
NEWLINE
VARIABLE 'W'
COLON ':'
INT 'int'
NEWLINE
VARIABLE 'H'
COLON ':'
INT 'int'
NEWLINE
RCURLY '}'
NEWLINE
LET 'let'
VARIABLE 'r'
EQUALS '='
VARIABLE 'rect'
LCURLY '{'
INTVAL '800'
COMMA ','
INTVAL '600'
RCURLY '}'
NEWLINE
SHOW 'show'
VARIABLE 'r'
DOT '.'
VARIABLE 'W'
NEWLINE
SHOW 'show'
VARIABLE 'r'
DOT '.'
VARIABLE 'H'
NEWLINE
END_OF_FILE
